cmd/data-usage-cache.go

```diff
@@ -774,15 +776,23 @@ func (d *dataUsageCache) save(ctx context.Context, store objectIO, name string)
 	// Abandon if more than 5 minutes, so we don't hold up scanner.
 	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
 	defer cancel()
-	_, err = store.PutObject(ctx,
-		dataUsageBucket,
-		name,
-		NewPutObjReader(r),
-		ObjectOptions{})
-	if isErrBucketNotFound(err) {
-		return nil
+	retry := true
+	for {
+		_, err = store.PutObject(ctx,
+			dataUsageBucket,
+			name,
+			NewPutObjReader(r),
+			ObjectOptions{})
+		if isErrBucketNotFound(err) || err == nil {
+			return nil
+		}
+		if err != errCorruptedFormat || !retry {
+			return err
+		}
+		// Clean up corrupted.
+		store.DeleteObject(ctx, dataUsageBucket, name, ObjectOptions{})
+		retry = false
```

Comment:
I am surprised shouldn't this be during Get instead of PUT? @klauspost

Reply:
@harshavardhana ok, I will leave it as a check for load and try breaking it again. That one I am positive about.